presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java

```diff
@@ -224,6 +226,44 @@ public static Double castToDouble(@SqlType(JSON) Slice json)
 
     @ScalarOperator(CAST)
     @SqlNullable
+    @SqlType(REAL)
+    public static Long castToFloat(@SqlType(JSON) Slice json)
+    {
+        try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
+            parser.nextToken();
+            Long result;
+            switch (parser.getCurrentToken()) {
+                case VALUE_NULL:
+                    result = null;
+                    break;
+                case VALUE_STRING:
+                    result = VarcharOperators.castToFloat(Slices.utf8Slice(parser.getText()));
+                    break;
+                case VALUE_NUMBER_FLOAT:
+                    result = (long) floatToRawIntBits(parser.getFloatValue());
+                    break;
+                case VALUE_NUMBER_INT:
+                    result = (long) floatToRawIntBits(parser.getFloatValue());
+                    break;
+                case VALUE_TRUE:
+                    result = (long) floatToRawIntBits(1);
```

Comment:
We should add `CAST BOOLEAN TO REAL` support and use `BooleanOperators.castToReal(true)`. Do this in a separate commit.